tests/ui/rfcs/rfc-2632-const-trait-impl/call-generic-method-nonconst.rs

```diff
@@ -20,7 +20,8 @@ const fn equals_self<T: ~const Foo>(t: &T) -> bool {
 // Calling `equals_self` with something that has a non-const impl should throw an error, despite
 // it not using the impl.
 
+// FIXME(effects): We should print `S: ~const Foo` here instead.
```

Comment:
Why? I believe that `S: const Foo` makes sense here, since this is in an always-const context...

Reply:
It does make sense to me as I've written above, I'm just not super keen on the
diagnostic note *required by this bound*, \*points at `~const`\*.
I can imagine ppl will open gh issues about that. Anyways.